packages/example-app-base/src/utilities/getSiteArea.ts

```diff
@@ -1,13 +1,23 @@
-import { pascalize } from './string';
+import { INavPage } from '../components/Nav/index';
+import { removeAnchorLink } from './removeAnchorLink';
 
 /**
- * Retrieves the current top level page name from the window URL or the passed hash.
+ * Retrieves the current top level page name defined in the pages array from the window URL or the passed hash.
+ * @param pages - Array of pages.
  * @param hash - The hash.
- * @param pascal - If the returned string should be in pascal case.
  */
-export function getSiteArea(hash?: string, pascal?: boolean): string {
-  hash = hash || window.location.hash;
-  const mod = pascal !== false;
-  const area = hash.indexOf('#/') > -1 ? hash.split('#/')[1].split('/')[0] : '';
-  return mod ? pascalize(area) : area;
+
+export function getSiteArea(pages?: INavPage[], hash: string = location.hash): string {
+  hash = removeAnchorLink(hash);
+  const topLevel = hash.indexOf('#/') > -1 ? hash.split('#/')[1].split('/')[0] : '';
```

Comment:
Some more comments explaining what this is doing would be good. Potentially also a test.